TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Users/Index.cshtml.cs

```diff
@@ -52,22 +46,22 @@ public async Task OnGetAsync()
                 ])
         ]);
 
-        var baseQuery = _dbContext.Users
+        var baseQuery = dbContext.Users
             .Where(u => u.UserType == UserType.Person && !string.IsNullOrWhiteSpace(u.Email))
             .Where(u => showAdminRole || u.Role != UserRoles.Administrator);
 
         var filteredQuery = filters.Apply(baseQuery);
 
-        var totalUserCount = await filters.CountAsync(filteredQuery);
```

Comment:
Not calling `filters.CountAsync()` will break the filter counts - surprised the tests pass

Reply:
Ah I see - I didn't realise `CountAsync()` had a side-effect

Reply:
`CountAsync()` is one of the main functions of the `Filters` class - I guess it could have a better name - maybe `CalculateFilterCounts` with an out parameter int `totalItemCount` - that would give more indication that the total count is more a by-product of a more complex operation